sdk/core/src/debug/java/com/klaviyo/core/Console.kt

```diff
@@ -0,0 +1,19 @@
+package com.klaviyo.core
+
+import android.util.Log
+
+/**
+ * Android Log output wrapper
+ */
+object Console {
+    enum class Level {
+        Debug, Info, Error, Assert
+    }
+
+    fun log(msg: String, level: Level, tag: String, ex: Throwable? = null) = when (level) {
+        Level.Debug -> ex?.let { Log.d(tag, msg, ex) } ?: Log.d(tag, msg)
+        Level.Info -> ex?.let { Log.i(tag, msg, ex) } ?: Log.i(tag, msg)
+        Level.Error -> ex?.let { Log.e(tag, msg, ex) } ?: Log.e(tag, msg)
+        Level.Assert -> ex?.let { Log.wtf(tag, msg, ex) } ?: Log.wtf(tag, msg)
```

Comment:
instead of doing this `when` statement on each log, you can reference the func as a property of the enum:
```
object Console {
    enum class Level(val call: (String, String, Throwable?) -> Unit) {
        Debug(Log::d), Info(Log::i), Error(Log::e), Assert(Log::wtf)
    }

    fun log(msg: String, level: Level, tag: String, ex: Throwable? = null) {
        level.call(tag, msg, ex)
    }
}
```

Reply:
based on source code, looks like they all handle nullable throwables so it should be fine

Reply:
oh dang thats a good idea